main.c

```diff
@@ -0,0 +1,721 @@
+            count++;
+            current = current->next;
+        }
+    }
+    
+    return sum / count;
+}
+
+
+void underperformedSstudents(struct School* school) {
+    for (int layer = 0; layer < MAX_LAYERS; layer++) {
+        for (int class = 0; class < MAX_CLASSES; class++) {
+            struct Student* current = school->layers[layer].classes[class].students;
+            while (current != NULL) {
+                if (averageGradesOfStudent(current) < 60) {
+                    printStudent(current);
+                }
+                current = current->next;
+            }
+        }
+    }
+}
+
+
+// Function to print a student's data to a file
+void printStudentToFile(FILE* file, struct Student* student) {
+
+    fprintf(file, "Layer: %d, Class: %d\n", student->gradeNumber, student->classNumber);
+    fprintf(file, "Student Name: %s %s\n", student->firstName, student->lastName);
+    fprintf(file, "Phone Number: %s\n", student->phoneNumber);
+    fprintf(file, "Grades: ");
+    for (int i = 0; i < 10; i++) {
+        fprintf(file, "%d ", student->grades[i]);
+    }
+    fprintf(file, "\n\n");
+}
+
+// Function to write all students' data to a new file
+void writeAllStudentsToFile(struct School* school, const char* filename) {
+    FILE* file = fopen(filename, "w");
+    if (file == NULL) {
+        printf("Error opening file!");
+        return;
+    }
+
+    int count = 0;
+    for (int layer = 0; layer < MAX_LAYERS; layer++) {
+        for (int class = 0; class < MAX_CLASSES; class++) {
+            struct Student* current = school->layers[layer].classes[class].students;
+            while (current != NULL) {
+                printStudentToFile(file, current);
+                current = current->next;
+                count++;
+            }
+        }
+    }
+
+    fclose(file);
+    printf("Data written to %s. Count of students: %d\n", filename, count);
+}
+
+
+void menu(struct School* school) {
+    char input;
+    // school.name = "schoolName";
+    do {
+        printf("\n|School Manager<::>Home|\n");
+        printf("--------------------------------------------------------------------------------\n");
+        //printf("Welcome to ( %s ) School!\nYou have inserted ( %zu ) students.\n\n", school.name, school.numOfStudents);
+        printf("\t[0] |--> Insert\n");//
+        printf("\t[1] |--> Delete\n");//
+        printf("\t[2] |--> Edit\n");//
+        printf("\t[3] |--> Search\n");//
+        printf("\t[4] |--> Show All\n");//
+        printf("\t[5] |--> Top 10 students per course\n");//
+        printf("\t[6] |--> Underperformed students\n");//
+        printf("\t[7] |--> Average per course\n");
+        printf("\t[8] |--> Export\n");
+        printf("\t[9] |--> Exit\n");
+        printf("\n\tPlease Enter Your Choice (0-9): ");
+        fflush(stdin);
+        input = getc(stdin);
+        //fflush(stdin);
+        //getc(stdin);
+        switch (input) {
+            case Insert:{
+                char firstName[50];
```

Comment:
The menu should just call functions. 
It is important to maintain modularity.

Reply:
Fixed it.